can_testbench.py

```diff
@@ -352,6 +364,216 @@ def closeEvent(self, event):
             logging.debug('Ignoring graph window close event')
             event.ignore()
 
+class Interface(enum.Enum):
+    slcan = 0
+    udp_multicast = 1
+
+SLCAN_BITRATES = ('10000', '20000', '50000', '100000', '125000', '250000', '500000', '750000', '1000000', '83300')
+
+class CanConfig():
+    """
+    Source of truth for current and allowed configs
+
+    Attributes:
+
+    """ 
+    def __init__(self):
+        super().__init__()
+        self.config = configparser.ConfigParser()
+        self.scriptDir = path.dirname(path.abspath(__file__))
+        self.configFile = path.join(self.scriptDir, 'can_testbench.ini')
+        self.selected = Interface.udp_multicast
+        self.dbcFile = path.join(self.scriptDir, '../envgo/dbc/testbench.dbc')
+        self.options = [
+            {'interface': Interface.slcan.name,
+            'channel': '/dev/tty.usbmodem3946375033311',
+            'bitrate': '500000',
+            'receive_own_messages': False},
+            {'interface': Interface.udp_multicast.name,
+            'channel': '239.0.0.1',
+            'port': '10000',
+            'receive_own_messages': False}
+        ]
+        self.initConfig()
+        
+    def initConfig(self):
+        if not path.isfile(self.configFile):
+            self.writeConfig()
+        else:
+            self.readConfig()
+            
+    def writeConfig(self):
+        self.config['General'] = {
+            'default_interface': self.selected.name,
+            'dbc_file': self.dbcFile
+        }
+        for interface in Interface:
+            self.config[interface.name] = self.options[interface.value]
+            with open(self.configFile, 'w') as configfile:
+                self.config.write(configfile)
+            
+    def readConfig(self):
+        self.config.read(self.configFile)
+        general = self.config['General']
+        if general.get('default_interface', None) is not None:
+            self.selected = Interface[general['default_interface']]
+        if general.get('dbc_file', None) is not None:
+            self.dbcFile = general['dbc_file']
+        for interface in Interface:
+            self.options[interface.value] = self.config[interface.name]
+        
+    def index(self) -> int:
+        return self.selected.value
+    
+    def setIndex(self, index: int):
+        self.selected = Interface(index)
+        
+    def setChannel(self, channel: str):
+        if 'channel' in self.options[self.index()]:
+            self.options[self.index()]['channel'] = channel
+        
+    def changeBitrate(self, index: int):
+        if 'bitrate' in self.options[self.index()]:
```

Comment:
You're right, it makes more sense to take the bitrate and just check to make sure it's in the tuple instead of taking an index.

Reply:
I used index originally because the SLCAN class has a list of valid bitrates that the bitrate must be on or it will error. I instinctively wanted to avoid a linear search and guarantee the value would always be valid so I used the index. However, A) I'll just use an enum to check if the bitrate is valid and B) The performance hit is negligible even if I didn't.